frontend/app/events/page.tsx

```diff
@@ -0,0 +1,19 @@
+import { getCurrentSession } from "@/actions/users.server";
+import PageLayout from "@/components/PageLayout";
+import { UserDisplayData } from "@/schemas/fe.schema";
+import { redirect } from "next/navigation";
+import EventsFrame from "@/components/EventsFrame";
+
+export default async function GroupsPage() {
+	// if user is not authenticated, redirect them to login
+	const userData: UserDisplayData | null = await getCurrentSession();
```

Comment:
you could use the userAuth check here if you'd like! with:

`import {withAuth} from "@/components/UserAuthCheck";`

`export default withAuth(GroupsPage);`